Fundamentals - Image and Imaging, Part 3 - "Imaging everything"

bellabf opened the pull request:
Hello, hello, hello,

I am happy to send you the part 3 of imaging everything in your merry way.
You can find this section in the ''extension-image.mdx".
This section was written by me, and it was based on this [outline](https://github.com/johko/computer-vision-course/discussions/83).

I left a note for the reviewers in the text itself because between our original issue and the outline, one of the sections was split into two, so I thought it might have some confusion there.

If it is not too bothersome, might I ask for @ATaylorAerospace to fact-check me in the sections regarding astronomical observations? I tried my best to understand the material and to put into more simple terms. But it is possible I might have made some mistakes, and I don't know enough about it to correct them myself.

As you usual, thank you so much for the feedback
imaging team concerned with this pr : @seshu-pavan 
bcc -  @lunarflu @merveenoyan @MKhalusova

I know @johko said he is not the code owner so we shouldn't wait for his reviews, but he is mentioned on the text due to his early input, so I am also tagging him here.

Let's goooo, I can't wait to get the course out there!!!

ATaylorAerospace commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+Event Horizon Telescope, 
+-->
+
+Another astronomically important event occurred in 2017 when researchers took the first picture of a black hole. It was the portrait of the mammoth black hole at the heart of the M87 galaxy. The image combined observations from multiple telescopes into one image. By combining their information, scientists can reconstruct the data from different sensors into one image. The data alone was more than ...
```
```suggestion
Another astronomically important event occurred in April 2019 when researchers captured the first image of a black hole! It was the image of the supermassive black hole at the center of the M87 galaxy in the constellation Virgo, about 55 million light-years away from Earth. The remarkable image was a product of the Event Horizon Telescope, a global network of synchronized radio observatories that worked together to create a virtual telescope as large as Earth. The data collected was enormous, over a petabyte, and had to be physically transported for processing due to its size. They needed to combine data coming from near-infrared, x-ray, millimeter wavelengths, and radio observations. This achievement was the culmination of years of effort by the Event Horizon Telescope Collaboration.
```

ATaylorAerospace replied:
@bellabf  I made a couple of updates to this paragraph to the dates and to give a bit more overview of  the Event Horizon  Telescope Collaboration Your overview was a GREAT foundation!

ATaylorAerospace commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+Event Horizon Telescope, 
+-->
+
+Now with that know-how, they aimed to image the supermassive black hole in the center of our galaxy, Sagittarius A*. Imaging Sagittarius A* was trickier compared to its counterpart. It is surrounded by a lot of dust and gas. Thus its environment makes it more difficult to get a clear photo. Researchers were worried that the environment would change too quickly to get a clear picture of what was g ...
```
```suggestion
Following the success with M87*, astronomers aimed to image the supermassive black hole at the center of our galaxy, Sagittarius A*. Imaging Sagittarius A* posed unique challenges due to its smaller size and the rapid variability in its surrounding environment, which changes much faster than the environment around larger black holes like M87*. This rapid movement made it difficult to capture a stable image that accurately represents the structure around Sagittarius A*. Just like our kitten example! Despite these challenges, the images obtained are significant for testing Einstein's theory of general relativity under extreme gravitational conditions. While these observations are crucial, they are part of a broader array of methods used to test the predictions of general relativity.
```

ATaylorAerospace replied:
@bellabf Same here I just added a bit more context to what you captured in terms of the imaging of Sagittarius A.  I really like how you mentioned that imaging the rapid movement of Sagittarius A is not a lot different than capturing imaging for a rapidly moving kitten!

ATaylorAerospace reviewed (commented):
I think you are ready for a second career as an astronomer @bellabf ! Your details on astronomical observations were very good and it was an excellent idea to focus on imaging from the Event Horizon Telescope in this section

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+ Example(s) of how image-acquired characteristics of images come from what types of sensors they record, challenges/difficulties in computer vision change according to the image/data, not all data is RGB
+
+-->
+
```
You can remove the parts above

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+##  Challenges in imaging real-life
+
+Have you ever tried to take a picture of a litter of kittens? If were not privileged with this experience, you are missing out on the most beautiful chaotic mess.  Kittens are adorable creatures that move around in the most deranged ways. They will do the cutest thing possible, but it will only last half a second before they top it off with an even cuter event. Before you know it, you are bending ...
```
```suggestion
Have you ever tried to take a picture of a litter of kittens? If you were not privileged with this experience, you are missing out on the most beautiful, chaotic mess.  Kittens are adorable creatures that move around in the most deranged ways. They will do the cutest thing possible, but it will only last half a second before they top it off with an even cuter event. Before you know it, you are bending yourself backward to get that one kitten in the frame while changing the zoom and the angle of the camera meanwhile, you have another kitten climbing your leg. You get so immersed in their fluffiness that you do not have the time to check the photos. When you sit down to check on them. They. are. all. just. a. blur. There is only one or two pictures worth keeping on your phone. You are just left there thinking, I thought kittens were more photogenic. 
```

johko replied:
@bellabf It says resolved but the text right now still looks like the old one and not like Merve's suggestion, maybe you forgot to change it?

bellabf replied:
> @bellabf It says resolved but the text right now still looks like the old one and not like Merve's suggestion, maybe you forgot to change it?

Hi, my bad. I don't know if I forgot or what happened, but it should be good now. :)

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+It is tempting to think that if we just had a better camera, one that responds more rapidly with a high resolution all would be solved. We would get the adorable pictures we want. Moreover, we will use the knowledge in this course to do more than just capture all of the adorable kittens, we will want to build a model on a nanny cam that checks if the kittens are still together with their mommy so ...
+
+Before we go out to buy the newest flashiest new camera in the market thinking we will have better data. It will be super easy to train. We will have a super-accurate model. Out-of-this-world performance on the kitten tracking market. This paragraph is here to guide you in a more productive direction and possibly save you a lot of time and money. A higher resolution is not the answer to all your  ...
```
```suggestion
Before we go out to buy the newest flashiest new camera in the market thinking we will have better data. It will be super easy to train a model. We will have a super-accurate model. Out-of-this-world performance on the kitten tracking market. This paragraph is here to guide you in a more productive direction and possibly save you a lot of time and money. A higher resolution is not the answer to all your problems. For starters, a typical neural network model for dealing with images is a convolution neural network (CNN). CNNs expect an image of a given size.  A large image needs a large model. Training will take a longer time. Chances are that your computers are also limited in RAM. A larger image size will mean fewer images to train on because the RAM will be limited for each iteration.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+It is not surprising that, as a species, we have developed new ways of seeing beyond the range of what our eyes could capture. We want to see what nature did not allow us to see in the first place. I can almost guarantee that if there is something out there that we are not sure of what it looks like, there is someone there trying to image it.
+
+As a species, we only see a fraction of the spectrum. We call that the visible spectrum. The image below shows us just how narrow it is: 
```
```suggestion
As human species, we only see a fraction of the spectrum. We call that the visible spectrum. The image below shows us just how narrow it is: 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+## Perspective on Imaging
+
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+
+## Perspective on Imaging
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+-->
+
+This diversity in imaging is quite phenomenal. These optical tools have become the eyes through which we perceive the universe. They provide us with insights that have revolutionized our understanding of the universe and life itself.  We use it on a daily basis to send pictures of our loved ones when they are away. We get an x-ray when the doctors need a closer look. Pregnant people have ultrasou ...
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+ The first picture of the DNA is also known as Photo 51. To image it, they use a technology based on fiber diffraction image of a crystal gel composed of DNA fiber. It was taken by Raymond Gosling, a graduate student working under the supervision of Rosalind Franklin in May 1952. It was a key piece of the double helix model constructed by Watson and Crick in 1953. There is a lot of controversy su ...
+
+### The pale blue dot
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+![The Pale Blue Dot](
+https://upload.wikimedia.org/wikipedia/commons/7/73/Pale_Blue_Dot.png)
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+-->
+
+The pale blue dot is a picture taken in 1990 by a space probe. Earth's size is so small that is less than a pixel. The picture received a lot of notoriety by showing how tiny and short Earth is relative to the vast majority of the space. It inspired Carl Sagan to write the book "The Pale Blue Dot". This picture was taken by the 1500mm high-resolution narrow-angle camera on the Voyager 1. The spac ...
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+
+### Black hole
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+### Black hole
+
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+https://upload.wikimedia.org/wikipedia/commons/thumb/4/4f/Black_hole_-_Messier_87_crop_max_res.jpg/1024px-Black_hole_-_Messier_87_crop_max_res.jpg)
+
+<!--
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+Another astronomically important event occurred in 2017 when researchers took the first picture of a black hole. It was the portrait of the mammoth black hole at the heart of the M87 galaxy. The image combined observations from multiple telescopes into one image. By combining their information, scientists can reconstruct the data from different sensors into one image. The data alone was more than ...
+
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+-->
+
+Another astronomically important event occurred in 2017 when researchers took the first picture of a black hole. It was the portrait of the mammoth black hole at the heart of the M87 galaxy. The image combined observations from multiple telescopes into one image. By combining their information, scientists can reconstruct the data from different sensors into one image. The data alone was more than ...
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+![Sag A](https://upload.wikimedia.org/wikipedia/commons/thumb/1/15/Sag_A%2A.jpg/1024px-Sag_A%2A.jpg)
+
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+
+<!--
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+
+![gif dna](https://cdn.technologyreview.com/i/images/horsegif_0.gif)
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+### Images, images, images
+
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,218 @@
+
+
+### Images, images, images
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+-->
+
+## Definition of Image
+An image is a visual representation of something. An image is an n-dimensional function.Tipically, we consider it to be two-dimensional (n=2). Let us denote: F(X, Y), where X and Y are spatial coordinates. The amplitude of F at a pair of coordinates (x1, y1) is the intensity or gray level of the image at that point. Typically, when we have the pair coordinate x1 and y1, we refer to them as pixels ...
```
```suggestion
An image is a visual representation of something. An image is an n-dimensional function. Typically, we consider it to be two-dimensional (n=2). Let us denote F(X, Y), where X and Y are spatial coordinates. The amplitude of F at a pair of coordinates (x1, y1) is the intensity or gray level of the image at that point. When we have the pair coordinate x1 and y1, we refer to them as pixels (picture elements).
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+## Definition of Image
+An image is a visual representation of something. An image is an n-dimensional function.Tipically, we consider it to be two-dimensional (n=2). Let us denote: F(X, Y), where X and Y are spatial coordinates. The amplitude of F at a pair of coordinates (x1, y1) is the intensity or gray level of the image at that point. Typically, when we have the pair coordinate x1 and y1, we refer to them as pixels ...
+
+Most of the images are generated by a combination of an illumination source and the reflection or absorption of energy. Images are discrete, the processes involved in assembling them are continuous. Image generating process will be discussed in greater detail in the following sections: technical aspects of imaging and how we imaged everything. 
```
```suggestion
Most of the images are generated by a combination of an illumination source and the reflection or absorption of energy. Images are discrete, and the processes involved in assembling them are continuous. The image generation process will be discussed in greater detail in the following sections: technical aspects of imaging and how we imaged everything. 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+Most of the images are generated by a combination of an illumination source and the reflection or absorption of energy. Images are discrete, the processes involved in assembling them are continuous. Image generating process will be discussed in greater detail in the following sections: technical aspects of imaging and how we imaged everything. 
+
+What matters to us right now is that the value of F at specific coordinates holds a physical meaning. And that the function F(X, Y) is characterized by two components: the amount of illumination from the source and the amount of illumination reflected by the object in the scene. Intensity images are also constrained in their intensity since the function is typically non-negative and their values  ...
```
```suggestion
What matters to us right now is that the value of F at specific coordinates holds a physical meaning. And that the function F(X, Y) is characterized by two components: the amount of illumination from the source and the amount of illumination reflected by the object in the scene. Intensity images are also constrained in their intensity since the function is typically non-negative, and their values are finite. Sometimes, when working with images, we might come across images with values smaller than zeros. This is sometimes done to signal the algorithm not to look at that part of the image. You can think about it as a sticker you put on an old photograph with your ex in it. 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+## Images vs other forms of data
+
+In tabular data, dimensionality is usually defined by the number of features (columns) describing one data point. In data, dimensionality usually refer to the cardinality of your combined dimensions. Due to the spatial relation in the data each pixel contributes to add more dimensions to the data. Features that describe your image are usually generated by traditional preprocessing or learned thro ...
```
```suggestion
In tabular data, dimensionality is usually defined by the number of features (columns) describing one data point. In data, dimensionality usually refers to the cardinality of your combined dimensions. Due to the spatial relation in the data, each pixel contributes to adding more dimensions to the data. Features that describe your image are usually generated by traditional preprocessing or learned through deep learning methods. Thus, feature extraction in an image involves different algorithms, which will be discussed in detail in the next chapter of this unit.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+If you have been paying attention, you might be guessing the videos are a visual representation of images that have a component of time attached to them. For 2D image acquistion, you would can add a time component such that F(X, Y, T). 
+
+Images can naturally have a hidden component in time but images and videos differ in how they sample this temporal information. An image is a static representation of a scene captured at a single point in time. While a video is a
```
You can remove the newline since it breaks the text

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+## Difference between Image and Video
+
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+## Difference between Image and Video
+
+
+If you have been paying attention, you might be guessing the videos are a visual representation of images that have a component of time attached to them. For 2D image acquistion, you would can add a time component such that F(X, Y, T). 
```
```suggestion
If you have been paying attention, you might guess the videos are a visual representation of images with a component of time attached to them. For 2D image acquisition, you can add a time component such that F(X, Y, T). 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+But how are they represented in the computer? Most commoly by matrices. It is easy to picture an image as a 2-D numerical array. This is an advantage because arrays are easy to deal with in computers. Apart from different performances in memory and reading, most different image formats end up in your script as a numpy array. Seeing matrices as images helps to understand some of the processes in c ...
+
+Alternatively, images might be represented as graphs where each edge is a coordinate and the edges are the neighboring coordinates. Take a moment to let that sink in, this means that the algorithms and models that are used for graphs can also be used for images. The inverse can also be true, you might be able to transform a graph into an image and analyze it as if it were a picture.
```
```suggestion
Alternatively, images might be represented as graphs where each edge is a coordinate, and the edges are the neighboring coordinates. Take a moment to let that sink in, this means that the algorithms and models used for graphs can also be used for images. The inverse can also be true, you might be able to transform a graph into an image and analyze it as if it were a picture.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+On occasion, it is also possible to hear that people are working on 4D or 5D images/ this terminology is mostly used by people in the biomedical field and microscopists. Again fear not! This naming came to be from people that image volumetric data in time, different channels, or different imaging modalities (i.e. photo and an x-ray). The idea is that each new source of information becomes an extr ...
+
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+On occasion, it is also possible to hear that people are working on 4D or 5D images/ this terminology is mostly used by people in the biomedical field and microscopists. Again fear not! This naming came to be from people that image volumetric data in time, different channels, or different imaging modalities (i.e. photo and an x-ray). The idea is that each new source of information becomes an extr ...
+
+
+But how are they represented in the computer? Most commoly by matrices. It is easy to picture an image as a 2-D numerical array. This is an advantage because arrays are easy to deal with in computers. Apart from different performances in memory and reading, most different image formats end up in your script as a numpy array. Seeing matrices as images helps to understand some of the processes in c ...
```
```suggestion
But how are they represented on the computer? Most commonly by matrices. It is easy to picture an image as a 2-D numerical array. This is an advantage because arrays are easy to deal with in computers. Apart from different performances in memory and reading, most different image formats end up in your script as a numpy array. Seeing matrices as images helps us understand some of the processes in convolution neural networks and image preprocessing, which we will see in later chapters. For instance, aligning an image might be decomposed into a mix of rotation and shearing operations, much like the typical linear algebra problem. 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+The most common color models is the RGB (Red, Green, Blue). Three colors, the channels, three F(X, Y). In this model, the true color of a coordinate (x, y) is given by the combination of the three different F(x,y) at that same location. Combined these three colors describe a wide spectrum of colors.
+
+If you only looking at the F(x, y) for one color, we called the gray level or the intensity.  The intensity values for each channel typically range from 0 to 255, where 0 represents no intensity and 255 represents maximum intensity. If you are in a different color system, these values might change so it is important to understand where your data come from. 
```
```suggestion
If you only look at the F(x, y) for one color, we call it the gray level or the intensity.  The intensity values for each channel typically range from 0 to 255, where 0 represents no intensity, and 255 represents maximum intensity. If you are in a different color system, these values might change, so it is important to understand where your data come from. 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+Sometime people will refer to different image channels. Fear not! Image channels are the different color components that make an image. In reference to the F(X,Y), this means we will have F for each color component.
+
+The most common color models is the RGB (Red, Green, Blue). Three colors, the channels, three F(X, Y). In this model, the true color of a coordinate (x, y) is given by the combination of the three different F(x,y) at that same location. Combined these three colors describe a wide spectrum of colors.
```
```suggestion
The most common color model is the RGB (Red, Green, Blue). Three colors, the channels, three F(X, Y). In this model, the true color of a coordinate (x, y) is given by combining the three different F(x,y) at that same location. Combining these three colors describes a wide spectrum of colors.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+A different time of image is volumetric or 3D images images. 3D images have the number of dimensions is equal to three and as a result have an F(X, Y, Z). Most of our reason still applies, with the only difference being that the triplet x1, y1, z1 is called a voxel.
+
+Sometime people will refer to different image channels. Fear not! Image channels are the different color components that make an image. In reference to the F(X,Y), this means we will have F for each color component.
```
```suggestion
Sometimes, people will refer to different image channels. Fear not! Image channels are the different color components that make an image. In reference to the F(X,Y), we will have F for each color component.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,77 @@
+
+What matters to us right now is that the value of F at specific coordinates holds a physical meaning. And that the function F(X, Y) is characterized by two components: the amount of illumination from the source and the amount of illumination reflected by the object in the scene. Intensity images are also constrained in their intensity since the function is typically non-negative and their values  ...
+
+A different time of image is volumetric or 3D images images. 3D images have the number of dimensions is equal to three and as a result have an F(X, Y, Z). Most of our reason still applies, with the only difference being that the triplet x1, y1, z1 is called a voxel.
```
```suggestion
A different type of image is volumetric or 3D images. 3D images have three dimensions. As a result, we have the image represented as F(X, Y, Z). Most of our reason still applies, with the only difference being that the triplet x1, y1, z1 is called a voxel.
```

merveenoyan reviewed (commented):
I mostly did grammatical corrections. Thank you!

merveenoyan commented:
@bellabf I saw some comments aren't addressed, can you address it and request for review again?

ATaylorAerospace reviewed (approved):
This unit looks great!

bellabf commented:
> @bellabf I saw some comments aren't addressed, can you address it and request for review again?

Hi, Merve!
I think I've addressed all the comments that were made previously :) 
Sending it back to you.

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+#  Imaging in real-life
```
```suggestion
#  Imaging in Real-life
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+
+Before we go out to buy the newest flashiest new camera in the market thinking we will have better data. It will be super easy to train a model. We will have a super-accurate model. Out-of-this-world performance on the kitten tracking market. This paragraph is here to guide you in a more productive direction and possibly save you a lot of time and money. A higher resolution is not the answer to a ...
+
+ I know that the evident solution is to say that we just get a computer with more GPU and more RAM. This also means that besides buying the camera, you will have to pay more for whatever service you will use to train the kitten model. More generally, this does not reflect real-world scenarios. Sometimes, the real application of a computer model is a GPU and memory-poor application. Wait, isn't th ...
```
```suggestion
The evident solution is to say that we just get a computer with more GPU and more RAM. This also means that besides buying the camera, you will have to pay more for whatever service you will use to train the kitten model. More generally, this does not reflect real-world scenarios. Sometimes, the real application of a computer model is a GPU and memory-poor application. Wait, isn't that our case in the first place? How are we going to fit our model into the hardware of the nanny can? 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+
+ I know that the evident solution is to say that we just get a computer with more GPU and more RAM. This also means that besides buying the camera, you will have to pay more for whatever service you will use to train the kitten model. More generally, this does not reflect real-world scenarios. Sometimes, the real application of a computer model is a GPU and memory-poor application. Wait, isn't th ...
+ 
+I got an idea, we will try a smaller model to have the same behavior as the big model! By the way,  this is an actual thing you can do. But even doing so, collecting the highest quality possible might not be a great idea simply because it usually takes longer to acquire and transmit it. A 50Gb of kitten pictures is still a 50Gb of data. No matter how adorable its contents are. Another argument is ...
```
```suggestion
We have an idea: we will try a smaller model to have the same behavior as the big model! By the way,  this is an actual thing you can do. But even doing so, collecting the highest quality possible might not be a great idea simply because it usually takes longer to acquire and transmit it. A 50Gb of kitten pictures is still a 50Gb of data. No matter how adorable its contents are. Another argument is that computer resources are usually either paid for or shared. In the first case, this might not be a good use of money resources. And as for the second, taking up an entire server is rarely a good way to make friends.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+ 
+I got an idea, we will try a smaller model to have the same behavior as the big model! By the way,  this is an actual thing you can do. But even doing so, collecting the highest quality possible might not be a great idea simply because it usually takes longer to acquire and transmit it. A 50Gb of kitten pictures is still a 50Gb of data. No matter how adorable its contents are. Another argument is ...
+
+There is even a better reason to not go for the highest resolution possible. The higher resolution might have more noise compared to a low one. Resolution amplifies not only your capability to capture the signal you are interested in but also your capability to pick up noise.  Thus, it might be easier to learn something on a lower-resolution image. Lower resolutions might help to have faster trai ...
```
```suggestion
There is even a better reason not to go for the highest resolution possible. The higher resolution might have more noise compared to a low one. Resolution amplifies not only your capability to capture the signal you are interested in but also your capability to pick up noise.  Thus, it might be easier to learn something on a lower-resolution image. Lower resolutions might help to have faster training, higher accuracy, and a cheaper model, computationally and monetaryally speaking. All of that being said, the takeaway here is to go for the highest resolution possible given the noise characteristics of the image and the infrastructure required both to train and deploy the model. And lastly, why are we using a high-quality camera in the first place? If we want to build a model on a nanny cam, we might as well get the pictures from the nanny cam.
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+I got an idea, we will try a smaller model to have the same behavior as the big model! By the way,  this is an actual thing you can do. But even doing so, collecting the highest quality possible might not be a great idea simply because it usually takes longer to acquire and transmit it. A 50Gb of kitten pictures is still a 50Gb of data. No matter how adorable its contents are. Another argument is ...
+
+There is even a better reason to not go for the highest resolution possible. The higher resolution might have more noise compared to a low one. Resolution amplifies not only your capability to capture the signal you are interested in but also your capability to pick up noise.  Thus, it might be easier to learn something on a lower-resolution image. Lower resolutions might help to have faster trai ...
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+
+![Image showing the visible spectrum compared to the Electromagnetic Spectrum](https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/human_spectrum.jpg)
+
+<!--
```
I think you can remove this part

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+We are so obsessive about seeing things that scientists have even changed the DNA sequence of certain animals so they can tag proteins of interest with a special type of protein, called green fluorescence protein. As the name suggests, when a green wavelength of light illuminates the sample, the GFP emits a fluorescent signal back.  Now, it is easier to know where the protein of interest is being ...
+
+After that, it was a matter of improving this system to get more channels in place, in a longer timescale, in a better resolution. A great example of this is how microscopes now generate terabytes of data overnight. 
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+
+![embryo image](https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/fish.gif)
+
+<!--
```
You can also remove this

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/Photo_51_x-ray_diffraction_image.jpg)
+
+
+<!--
```
You can remove this

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+
+-->
+
+ The first picture of the DNA is also known as Photo 51. To image it, they use a technology based on fiber diffraction image of a crystal gel composed of DNA fiber. It was taken by Raymond Gosling, a graduate student working under the supervision of Rosalind Franklin in May 1952. It was a key piece of the double helix model constructed by Watson and Crick in 1953. There is a lot of controversy su ...
```
```suggestion
The first picture of the DNA is also known as Photo 51. They use a technology based on fiber diffraction image of a crystal gel composed of DNA fiber to image it. It was taken by Raymond Gosling, a graduate student working under the supervision of Rosalind Franklin in May 1952. It was a key piece of the double helix model constructed by Watson and Crick in 1953. There is a lot of controversy surrounding this photo. Part of it comes from the unrecognized contribution made by Rosalind Franklin's early work and the circumstances under which the photo was shared with Watson and Crick. Nevertheless, it has significantly contributed to our understanding of DNA's structure and the technologies that were developed thereafter. 
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/Pale_Blue_Dot.png)
+
+
+<!--
```
Also this

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+The pale blue dot is a picture taken in 1990 by a space probe. Earth's size is so small that is less than a pixel. The picture received a lot of notoriety by showing how tiny and short Earth is relative to the vast majority of the space. It inspired Carl Sagan to write the book "The Pale Blue Dot". This picture was taken by the 1500mm high-resolution narrow-angle camera on the Voyager 1. The spac ...
+
+
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+
+The pale blue dot is a picture taken in 1990 by a space probe. Earth's size is so small that is less than a pixel. The picture received a lot of notoriety by showing how tiny and short Earth is relative to the vast majority of the space. It inspired Carl Sagan to write the book "The Pale Blue Dot". This picture was taken by the 1500mm high-resolution narrow-angle camera on the Voyager 1. The spac ...
+
+
```
```suggestion
```

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+![M87](
+https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/Black_hole_-_Messier_87.jpg)
+
+<!--
```
Also this

merveenoyan commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,161 @@
+<!--
+Event Horizon Telescope, 
+-->
+
```
```suggestion
```

merveenoyan reviewed (commented):
I think we can merge this next. You can have these texts inside arrows in alt text inside the image or at the bottom of the page

bellabf commented:
> I think we can merge this next. You can have these texts inside arrows in alt text inside the image or at the bottom of the page

Hi, again! 
I've added the text inside the arrow into the alt text, it should be good to go :)

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,102 @@
+
+Before we go out to buy the newest flashiest new camera in the market thinking we will have better data. It will be super easy to train a model. We will have a super-accurate model. Out-of-this-world performance on the kitten tracking market. This paragraph is here to guide you in a more productive direction and possibly save you a lot of time and money. A higher resolution is not the answer to a ...
+
+The evident solution is to say that we just get a computer with more GPU and more RAM. This also means that besides buying the camera, you will have to pay more for whatever service you will use to train the kitten model. More generally, this does not reflect real-world scenarios. Sometimes, the real application of a computer model is a GPU and memory-poor application. Wait, isn't that our case i ...
```
```suggestion
The evident solution is to say that we just get a computer with more GPU and more RAM. This also means that besides buying the camera, you will have to pay more for whatever service you will use to train the kitten model. More generally, this does not reflect real-world scenarios. Sometimes, the real application of a computer model is a GPU and memory-poor application. Wait, isn't that our case in the first place? How are we going to fit our model into the hardware of the nanny cam? 
```

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,102 @@
+](https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/human_spectrum.jpg)
+
+
+To see more than what Mother Nature has given us, we need sensor capturing beyond that spectrum. In other words, we need to detect things at different wavelengths.  Infrared (IR) is used in night vision devices and some astronomical observations. Magnetic resonance uses strong magnetic fields and radio waves to image soft human tissues. We created ways to see things that do not rely on light. For ...
```
```suggestion
To see more than what Mother Nature has given us, we need sensors capturing beyond that spectrum. In other words, we need to detect things at different wavelengths.  Infrared (IR) is used in night vision devices and some astronomical observations. Magnetic resonance uses strong magnetic fields and radio waves to image soft human tissues. We created ways to see things that do not rely on light. For instance, electron microscopy uses electrons to zoom in at much higher resolution than traditional light.  Ultrasound is another great example. Ultrasound imaging harnesses sound waves to create detailed, real-time images of internal organs and tissues, offering a non-invasive and dynamic perspective that goes beyond what is achievable with standard light-based imaging methods.
```

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```diff
@@ -0,0 +1,102 @@
+
+After that, it was a matter of improving this system to get more channels in place, in a longer timescale, in a better resolution. A great example of this is how microscopes now generate terabytes of data overnight. 
+
+A great example of this combined effort is the video below. In it, you see the time lapse of the projection of the 3D image of a developping embryo of a fished tagged a fluorescent protein. Each colored dot you see on the image represent an individual cell.
```
```suggestion
A great example of this combined effort is the video below. In it, you see the time lapse of the projection of the 3D image of a developping embryo of a fished tagged a fluorescent protein. Each colored dot you see on the image represents an individual cell.
```

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
```suggestion
Understanding image characteristics is a really good first step in building a computer vision model. Not only because it will influence the performance of the computer vision model, but because it will dictate what models are more suitable for your problem. Notably, not every image type requires the development of a new neural network architecture. Sometimes, you can adapt a pre-existing model by fine-tuning it or manipulating the last layer to do a different task. Sometimes this manipulation is not needed; instead, preprocessing is employed to make your image more similar to the input that the network was trained on. Do not worry too much about the details of this right now, they will be addressed in the latter chapter of this course. They are mentioned here to help you understand why the context in which an image is acquired is relevant.
```

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/extension-image.mdx:
I cannot exactly say how to correct it, but the second part of the second sentence - `are imaged under you have a polar grid` seems grammatically wrong.

bellabf replied:
> I cannot exactly say how to correct it, but the second part of the second sentence - `are imaged under you have a polar grid` seems grammatically wrong.

I rewrote this and the next two sentences for clarity. It was a bit of an awkward sentence, indeed.

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,69 @@
+
+## Definition of Image
+
+An image is a visual representation of an object, a scence, a person, or even a concept They can be photographs, paintings, drawings, schemes, scans and more! One the more suprising things is that an image is also a function. More precisely, an image is an n-dimensional function. We will first consider it to be two-dimensional $(n=2)$ first. We will call it $F(X, Y)$ , where $X$ and $Y$ are spati ...
```
```suggestion
An image is a visual representation of an object, a scene, a person, or even a concept. They can be photographs, paintings, drawings, schemes, scans and more! One of the more surprising things is that an image is also a function. More precisely, an image is an n-dimensional function. We will first consider it to be two-dimensional $(n=2)$. We will call it $F(X, Y)$ , where $X$ and $Y$ are spatial coordinates. Do not get distracted by the fancy name. Spatial coordinates are just the system that we use to describe the positions of objects in a physical space with the most common one being the $2D$ Cartesian system. The amplitude of $F$ at a pair of coordinates $(x_i, y_i)$ is the intensity or gray level of the image at that point. The intensity is what gives you the perception of light and dark. Typically, when we have the coordinate pair $x_1$ and $y_1$, we refer to them as pixels (picture elements).
```

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,69 @@
+
+Images are discrete, and the processes involved in assembling them are continuous. The image generation processes will be discussed in  the next chapter. Right now what matters is that the value of $F$ at specific coordinates holds a physical meaning. And that the function $F(X, Y)$ is characterized by two components: the amount of illumination from the source and the amount of illumination refle ...
+
+That is not the only way one can create an image.  Sometimes, they are created by computers with or without the help of AI.  We have dedicated chapter for images that do have a little helping hand from AI. Most of the terminology, we will introduce here will still be apply
```
```suggestion
That is not the only way one can create an image.  Sometimes, they are created by computers with or without the help of AI.  We have a dedicated chapter for images that do have a little helping hand from AI. Most of the terminology we will introduce here will still be applicable.
```

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,69 @@
+
+If you're only looking at the $F(x, y)$ for one color, it ranges from $0$ to $255$, where $0$ represents no intensity and $255$ represents maximum intensity. In a different color system, combining these values might differ. So, it is important to understand where your data comes from when interpretating these values.
+
+There are special types of images where the coordinates $x_i, y_i$ do not describe an intensity value, but instead label a pixel. The simplest example of an operation that results in such an image is separating foreground and background. Everything that is foreground receives the label $1$, and everything that is background receives the label $0$. These images are commonly referred to as labeled  ...
```
```suggestion
There are special types of images where the coordinates $x_i, y_i$ do not describe an intensity value, but instead label a pixel. The simplest example of an operation that results in such an image is separating foreground and background. Everything that is foreground receives the label $1$, and everything that is background receives the label $0$. These images are commonly referred to as labeled images. When there are only two labels, like our example, we call them binary images or masks. 
```

johko commented on chapters/en/Unit 1 - Fundamentals/image_and_imaging/image.mdx:
```diff
@@ -0,0 +1,69 @@
+
+In tabular data, dimensionality is usually defined by the number of features (columns) describing one data point. In visual data, dimensionality usually refers to the number of dimensions that describe your data. For a $2D$ image, we usually refer to numbers $x_i$ and $y_i$ as the image size.  
+
+ Another aspect is the generation of features that describe visual data are generated by traditional preprocessing or learned through deep learning methods. We refer to this as feature extraction. It involves different algorithms discussed in more detail in the feature extraction chapter. It contrasts with the feature engineering for tabular data, where new features are built upon existing ones.
```
```suggestion
 Another aspect is the generation of features that describe visual data. They are generated by traditional preprocessing or learned through deep learning methods. We refer to this as feature extraction. It involves different algorithms discussed in more detail in the feature extraction chapter. It contrasts with the feature engineering for tabular data, where new features are built upon existing ones.
```

johko reviewed (commented):
Really great work, could as well be part of a book ;) 
Left some grammar corrections, but apart from that LGTM :+1:

bellabf commented:
> Really great work, could as well be part of a book ;) Left some grammar corrections, but apart from that LGTM 👍

Thank you!
Everyone's feedback (@merveenoyan  @ATaylorAerospace ) has been really encouraging. Thanks a lot. It means a lot to me :smile:

johko reviewed (approved):
Thanks for the quick changes :)
